jdbc/src/main/java/com/interface21/transaction/support/TransactionSynchronizationManager.java

```diff
@@ -2,22 +2,45 @@
 
 import javax.sql.DataSource;
 import java.sql.Connection;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.NoSuchElementException;
 
 public abstract class TransactionSynchronizationManager {
 
-    private static final ThreadLocal<Map<DataSource, Connection>> resources = new ThreadLocal<>();
+    private static final ThreadLocal<Map<DataSource, Connection>> resources = ThreadLocal.withInitial(HashMap::new);
 
-    private TransactionSynchronizationManager() {}
+    private TransactionSynchronizationManager() {
+    }
 
     public static Connection getResource(DataSource key) {
+        Map<DataSource, Connection> connections = resources.get();
+
+        if (connections.containsKey(key)) {
+            return connections.get(key);
+        }
         return null;
```

Comment:
이렇게 개선할 수 있어요! key가 없으면 null을 반환합니다.
```suggestion
        return connections.get(key);
```